### DISCUSSION TITLE
Resize a dicom image

### DISCUSSION THREAD

**Discussion by vikashg** [Project-MONAI/MONAI#3591] [General]
Hello everyone, 
I have dicom images which when read using the LoadImage() transform gives me an image of format (1, 1890, 2050). It is a 2D image. The first axis is the channel. However, I would like it to reshape it as an image of size (500, 500). 
**I was wondering if there was a transform to remove the first channel ?**
In addition, when I use the `Resize` filter as `Resize((1, 500, 500))` , it will throw an image of size (1, 1890, 500, 500). 
I do not understand this Resize function. 
Any help would be really appreciated.

**Comment by Nic-Ma** [comment]
Hi @vikashg ,

You can use the `SqueezeDim` transform:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utility/array.py#L509
And BTW, the MONAI API doc is: https://docs.monai.io/en/stable/api.html

Thanks.

**Comment by wyli** [comment]
The first parameter to `Resize` is a 'spatial' size, so in this case it would be `(500, 500)`: 
```py
monai.transforms.Resize((500, 500))(torch.zeros((1, 1890, 2050))).shape
```
https://docs.monai.io/en/stable/transforms.html#monai.transforms.Resize

**Reply by vikashg** [reply]
Thanks @wyli and @Nic-Ma , Although this works, I also tried something like 
```
monai.transforms.Resize((500, 500))(torch.zeros((1890, 2050, 1))).shape
```
It yields a shape (1890, 500, 500). The part which is confusing is that the position of channels affect the size of the output.

**Comment by vikashg** [comment]
In addition, 
if I do the following chain of transforms 
```
train_transforms = Compose([LoadImage(image_only=True, reader="itkReader"), SqueezeDim(dim=2),AddChannel(), Resize((500,500)), 
 EnsureType()])
```
I get an output of the size(1, 1, 500, 500), where the original image was of size (1890, 2050, 1). So I had to do this 
```
train_transforms = Compose([LoadImage(image_only=True, reader="itkReader"), SqueezeDim(dim=2),AddChannel(), Resize((500,500)), SqueezeDim(dim=0), EnsureType()])
```
to get an image of size (1, 500, 500). I just find the way `Resize()` is implemented a bit weird. Am I missing something in my implementation?

Thanks once again.

**Reply by Nic-Ma** [reply]
You can try the `EnsureChannelFirst` transform or `AsChannelFirst` instead of current logic.
All the MONAI transforms expect `channel-first` data shape.

Thanks.

**Reply by vikashg** [reply]
Ok so I tried this 
```
train_transforms = Compose([LoadImage(image_only=True, reader="itkReader"), AsChannelFirst(), Resize((500,500)),						EnsureType()])
```

The output is of the shape (1, 1, 500, 500). I feel the extra channel at the end is for the index in the batch of the dataset.

**Reply by Nic-Ma** [reply]
So is there still any problem in your test now?

Thanks.

**Reply by rijobro** [reply]
> The output is of the shape (1, 1, 500, 500). I feel the extra channel at the end is for the index in the batch of the dataset.

Our transforms don't add a batch dimension (this is only added by the `DataLoader`). I would suggest confirming the initial shape of your image with:

`print(LoadImage(image_only=True)(filename).shape)`

**Reply by vikashg** [reply]
Ok got it, Thanks @Nic-Ma, It is good now so far. I have the output shapes I was looking for. Thanks @rijobro  for the pointers.
